samples/readSnippets.js

```diff
@@ -0,0 +1,217 @@
+      // [END bigtable_reads_row_partial]
+      break;
+    }
+
+    // [START bigtable_reads_rows]
+    case 'readRows': {
+      const rowKeys = ['phone#4c410523#20190501', 'phone#4c410523#20190502'];
+
+      const [rows] = await table.getRows({keys: rowKeys});
+      for (let i = 0; i < rows.length; i++) {
+        printRow(rows[i].id, rows[i].data);
+      }
+      // [END bigtable_reads_rows]
+      break;
+    }
+
+    // [START bigtable_reads_row_range]
+    case 'readRowRange': {
+      const start = 'phone#4c410523#20190501';
+      const end = 'phone#4c410523#201906201';
+
+      await table
+        .createReadStream({
+          start,
+          end,
+        })
+        .on('error', err => {
+          // Handle the error.
+          console.log(err);
+        })
+        .on('data', row => printRow(row.id, row.data))
+        .on('end', () => {
+          // All rows retrieved.
+        });
+      // [END bigtable_reads_row_range]
+      break;
+    }
+
+    // [START bigtable_reads_row_ranges]
+    case 'readRowRanges': {
+      await table
+        .createReadStream({
+          ranges: [
+            {
+              start: 'phone#4c410523#20190501',
+              end: 'phone#4c410523#20190601',
+            },
+            {
+              start: 'phone#5c10102#20190501',
+              end: 'phone#5c10102#20190601',
+            },
+          ],
+        })
+        .on('error', err => {
+          // Handle the error.
+          console.log(err);
+        })
+        .on('data', row => printRow(row.id, row.data))
+        .on('end', () => {
+          // All rows retrieved.
+        });
+      // [END bigtable_reads_row_ranges]
+      break;
+    }
+
+    // [START bigtable_reads_prefix]
+    case 'readPrefix': {
+      const prefix = 'phone#';
+
+      await table
+        .createReadStream({
+          prefix,
+        })
+        .on('error', err => {
+          // Handle the error.
+          console.log(err);
+        })
+        .on('data', row => printRow(row.id, row.data))
+        .on('end', () => {
+          // All rows retrieved.
+        });
+      // [END bigtable_reads_prefix]
+      break;
+    }
+
+    // [START bigtable_reads_filter]
+    case 'readFilter': {
+      const filter = {
+        value: /PQ2A.*$/,
+      };
+
+      await table
+        .createReadStream({
+          filter,
+        })
+        .on('error', err => {
+          // Handle the error.
+          console.log(err);
+        })
+        .on('data', row => printRow(row.id, row.data))
+        .on('end', () => {
+          // All rows retrieved.
+        });
+      // [END bigtable_reads_filter]
+      break;
+    }
+  }
+
+  // [START bigtable_reads_row]
+  // [START bigtable_reads_row_partial]
+  // [START bigtable_reads_rows]
+  // [START bigtable_reads_row_range]
+  // [START bigtable_reads_row_ranges]
+  // [START bigtable_reads_prefix]
+  // [START bigtable_reads_filter]
+  function printRow(rowkey, rowData) {
```

Comment:
This could absolutely be correct, so please don't block on this comment.  Are we effectively trying to iterate over the entire object structure, and just log everything?  If so, something like this could suffice:

```js
console.log(util.inspect(row, { depth: 10 }));
```

Again - what you did is probably better, just throwing an alternative approach out there
```

Reply:
I'm trying to print it in a particular way to give people a clear reference of how the object is laid out and how to find the data they are looking for